service/controller/resource/promremotewrite/create.go

```diff
@@ -0,0 +1,75 @@
+package promremotewrite
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/resourcecanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "reconcile remotewrite")
+	{
+		// get remotewrite
+		remoteWrite, err := ToRemoteWrite(obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		selector, err := metav1.LabelSelectorAsSelector(&remoteWrite.Spec.ClusterSelector)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		// fetch current prometheus
+		prometheusList, err := r.prometheusClient.
+			MonitoringV1().
+			Prometheuses(metav1.NamespaceAll).
+			List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
+		if err != nil {
+			return microerror.Maskf(errorFetchingPrometheus, "Could not fetch Prometheus with label selector '%#q'", remoteWrite.Spec.ClusterSelector.String())
+		}
+		if prometheusList == nil && len(prometheusList.Items) == 0 {
+			r.logger.Debugf(ctx, "no prometheus found, cancel reconciliation")
+			resourcecanceledcontext.SetCanceled(ctx)
+			return nil
+		}
+
+		// loop
```

Comment:
Can you explain the purpose of this comment ?

Reply:
I just commented because I wanted to remember what I needed to do! so nothing special.

Reply:
)